### ISSUE TITLE
[Feature Request]: fold batch norm into convolution weights of graph transform tool 

### ISSUE THREAD

**Issue by ydzhang12345** [tensorflow/tensorflow#12740]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.3.0
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.5.1
- **CUDA/cuDNN version**: 8.0, 6.0
- **GPU model and memory**: GTX1080
- **Exact command to reproduce**:

bazel build tensorflow/tools/graph_transforms:transform_graph
bazel-bin/tensorflow/tools/graph_transforms/transform_graph \
--in_graph=tensorflow_inception_graph.pb \
--out_graph=optimized_inception_graph.pb \
--inputs='Mul' \
--outputs='softmax' \
--transforms='
  strip_unused_nodes(type=float, shape="1,299,299,3")
  remove_nodes(op=Identity, op=CheckNumerics)
  fold_constants(ignore_errors=true)
  fold_batch_norms
  fold_old_batch_norms'

After diving into the generated pb graph, I found that the current fold batch norm ops seems to only fold batch norm into two ops: mul and add, which is not exactly as the readme demonstrated: fold the batch norm into the weights of convolution. If this is the case, it would be better to further merge the mul and add into conv weights.

**Comment by carlthome** [comment]
On a similar note, what about folding [this](https://www.tensorflow.org/api_docs/python/tf/contrib/rnn/LayerNormBasicLSTMCell) as well? I.e. fold across `array_ops.split`.

**Comment by passerbyd** [comment]
fold_batch_norms/fold_old_batch_norms merges the scale part of a (fused) batch normalization into conv_op only if the bn op is trailing a conv op. Otherwise nothing is did.
And the tool run as expected in this case. You can check its implementation.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by ydzhang12345** [comment]
resolved

**Comment by MrWanter** [comment]
@ydzhang12345  could you pls share how do you resolve the problem?

**Comment by Toory465** [comment]
ANy Update for that? Iwanna know if you remove the batchnorm how do you finetune the network?

**Comment by ydzhang12345** [comment]
We manually optimize these in C.
